### PR TITLE
feat: Vital Content Listing

### PR DESCRIPTION
**Pull request by wodenx**
## Changes
- Create vital content listing package and content listing variation of generic template
- Small changes to bodielss filtering: Query parameters do not override default filters
- 
## Test Instructions
There are 3 stleguide pages
- content-listing - uses site-wide data, single allowed filter, no default filters
- content-listing-2 - use site-wide data, single alllowed filter, default filter set
- contont-listing-multiple - uses page-local data, multiple allowed filters, no default filter selector
Also - the styleguide page is replaced with a content listing page using site-wide data.

To verify:
- reset button is removed and fitlers are disbled when a default filter is specified
- query parameters are  overridden when a default filter is specified (they should be set to what is defined by the default filter)
-

### CODE REVIEW

File: packages/__vital__/src/styleguide/Page/StyleGuideTemplate/ContentListing.tsx [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+import { asStyleGuideTemplateToken, vitalStyleGuideTemplate } from '@bodiless/vital-templates';
+import {
+  as, flowHoc, replaceWith,
+} from '@bodiless/fclasses';
+import { ContentListingClean, vitalContentListing } from '@bodiless/vital-content-listing';
+
+const Default = as(
+  vitalContentListing.Default,
+  vitalContentListing.WithFilterSelector,
+  withNodeKey({ nodeKey: 'styleguide-content-listing', nodeCollection: 'site' }),
```
Comment (dewen) [inline packages/__vital__/src/styleguide/Page/StyleGuideTemplate/ContentListing.tsx:26]: do we need this site level node key?

File: sites/vital-demo/src/data/pages/styleguide/content-listing-multiple/description.json [hunk 2]
```diff
@@ -0,0 +1,10 @@
+[
+  {
+    "type": "paragraph",
+    "children": [
+      {
+        "text": "This page demonstrates the content listing page with multiple allowed filters.  The filters and content are local to this page.."
```
Comment (dewen) [inline sites/vital-demo/src/data/pages/styleguide/content-listing-multiple/description.json:6]: remove extra period.
` this page..` => `this page.`

File: sites/vital-demo/src/data/pages/styleguide/content-listing-2/description.json [hunk 3]
```diff
@@ -0,0 +1,27 @@
+[
+  {
+    "type": "paragraph",
+    "children": [
+      {
+        "text": "This page demonstrates the content listing filters with . single allowed tag.  The content and filters are site-wide.  This page has a default filter set.  Another view on the same content (with no default filter) can be found at  "
```
Comment (dewen) [inline sites/vital-demo/src/data/pages/styleguide/content-listing-2/description.json:6]: `with . single allowed tag` => `with single allowed tag`

File: packages/vital-content-listing/src/components/FilterByGroup/tokens/constants.ts [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+export enum FilterByGroupNodeKeys {
+  Filter = 'filter',
+  Content = 'content',
+  FiterSelector = 'filter-selector',
```
Comment (dewen) [inline packages/vital-content-listing/src/components/FilterByGroup/tokens/constants.ts:18]: `FiterSelector` => `FilterSelector`

File: packages/vital-content-listing/src/components/FilterByGroup/tokens/vitalFilterByGroup.ts [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+import { asFilterByGroupToken } from '../FilterByGroupClean';
+import { FilterByGroupNodeKeys } from './constants';
+
+/**
+ * Peek into the default filter node
+ */
+const useHasFilterDelection = () => {
+  const { node } = useNode('_default');
+  const { isEdit } = useEditContext();
+  const fsNode = node.child<any>(FilterByGroupNodeKeys.FiterSelector);
```
Comment (dewen) [inline packages/vital-content-listing/src/components/FilterByGroup/tokens/vitalFilterByGroup.ts:33]: `FiterSelector` => `FitlerSelector`

File: packages/vital-content-listing/src/components/FilterByGroup/tokens/vitalFilterByGroup.ts [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+  return fsNode.data.tags?.length && !isEdit;
+};
+
+const WithFilterSelector = asFilterByGroupToken({
+  Core: {
+    // Hide reset button if default filters are set.
+    ResetButton: flowIf(useHasFilterDelection)(replaceWith(() => null)),
+    Filter: withFilterSelection({
+      nodeKey: FilterByGroupNodeKeys.FiterSelector,
```
Comment (dewen) [inline packages/vital-content-listing/src/components/FilterByGroup/tokens/vitalFilterByGroup.ts:42]: `FiterSelector` => `FilterSelector`

File: packages/__vital__/src/styleguide/Page/StyleGuideTemplate/ContentListing.tsx (hunk 1 above)
Comment (wodenx) [reply packages/__vital__/src/styleguide/Page/StyleGuideTemplate/ContentListing.tsx:26]: yes - this is what instructs it to save the data at site level (shared data).

**Review by dewen — DISMISSED** [review]

**Review by dewen — APPROVED** [review]

**Review by DavtyanH — APPROVED** [review]

**Review by marcopagliarulo — APPROVED** [review]

**Pull request merged**